### DISCUSSION TITLE
Weblate and Python project: New strings commited in weblate repo but ignored in translation interface

### DISCUSSION THREAD

**Discussion by buhtz** [WeblateOrg/weblate#9173] [Q&A] [answered]
This is partly an X-Post from here: https://codeberg.org/Codeberg/Community/issues/965

As you can see in this screenshot I was able to keep my upstream repo and the weblate's internal repo synced. They are on the same state/commit.
![image](https://user-images.githubusercontent.com/11861496/234819150-578d314c-e078-48ac-ad72-ba2be21d07b4.png)

There are [two strings](https://codeberg.org/buhtz/bit_demo/src/commit/ababb45b52d6e02539332bff6b19eb1cb18175d1/src/bitcli/__main__.py#L50-L51) new strings in that repo that are not translated yet. But this strings do not appear in the translation interface here:

![image](https://user-images.githubusercontent.com/11861496/234818465-1e5eb32c-01af-4058-8deb-b6c178dc9379.png)

You see 5 strings. But there should be 7 with 2 untranslated.
How can this be when there are definitely (?) 7 translatable strings in weblates internal repo?

**Comment by nijel** [comment]
For some reason, Weblate didn't process the update. I can think of two reasons:

1. The process crashed, either due to a bug or being killed by OOM. This should be visible from the server logs. Having the logs showing the issue might be useful to diagnose.
2. The update of files was missed for some reason, but tracked as parsed by Weblate.

To recover, you will need console access to the Weblate server:

1. Try `weblate loadpo --foreground bitdemo/bit_demo`. If this does the update, the first issue happened. (In future, you will be able to do this from Weblate as well, I've just implemented UI for this in 804f9f9848aaa9a3e069ae697a7d582454cafeae).
2. Try `weblate loadpo --force --foreground bitdemo/bit_demo`. This would indicate the second issue.

**Reply by buhtz** [reply]
I don't have access to the server.

But I have an idea what you mean by "The update of files was missed for some reason".
The "two new strings in that repo" do exist in the py-files but not in the po-files. As a user I expect Weblate to do the extraction of translateable strings into po-files. But I assume this doesn't happen in Weblate?

**Reply by nijel** [reply]
Okay, if they do not exist in the PO files, we have FAQ covering this: https://docs.weblate.org/en/latest/faq.html#why-does-weblate-still-show-old-translation-strings-when-i-ve-updated-the-template

**Reply by buhtz** [reply]
That is what I meant. The toolchain of "pygettext3" and "msgmerge". Weblate should do this by itself because it has the whole repo cloned.

**Reply by nijel** [reply]
You can configure Weblate to do it.

**Reply by buhtz** [reply]
Nice to know. How can I configure it?

**Reply by pietervdvn** [reply]
I had a similar problem with `XXX strings are not being translated here` (or `XXX strings are not translated here`) under a little ghost icon + a new language not being translateable. The `loadpo --foreground <component>` worked for me.

(I'm adding the error message here so that this conversation can be discovered more easily)

**Comment by nijel** [comment]
As this has been recently commented on anyway, let's update the discussion:

* [Update POT file (xgettext)](https://docs.weblate.org/en/latest/admin/addons.html#update-pot-file-xgettext) add-on can be used to update POT files based on the source changes
* [Update PO files to match POT (msgmerge)](https://docs.weblate.org/en/latest/admin/addons.html#update-po-files-to-match-pot-msgmerge) can be used to update PO files based on POT

**Comment by buhtz** [comment, marked as answer]
Ui, that is an old thread. And it is out-dated.

I never used the mentioned add-ons, but found my own semi-automatic solution. After I understand that the Weblate instance does manage its on git repo of my upstream repo and I can access it with git commands, the solution was obvious.

The result is [this script](https://github.com/bit-team/backintime/blob/dev/update_language_files.py) that manage to synchronize the upstream repo with the Weblate instance. Additionally it does a lot more syntax checking and other stuff related to my specific project.
